### DISCUSSION TITLE
products_mpn field

### DISCUSSION THREAD

**Discussion by piloujp** [zencart/zencart#6806] [General]
Did I miss something, or is the field `products_mpn` in products table not editable at all in admin?
It seems the only way to edit it in admin is to add some custom code or use the plugin Extra Product Field.

**Comment by torvista** [comment]
You missed the party:
https://github.com/zencart/zencart/issues/5727

Addition
https://github.com/zencart/zencart/pull/5893/files

**Comment by piloujp** [comment]
Thanks for the link, I effectively missed this.
But after reading it all, I still don't understand why this field was added to `products` table if it is to do nothing with it... Why the collect_info files have not been modified to make it editable in admin products edit pages?

**Reply by drbyte** [reply]
It was done as merely a schema update so that the various plugins that use it wouldn't have to do any database updates to add the field.

Did you arrive at noticing its presence because you wanted to use it? Or merely by observing that the field existed in the database?

**Reply by piloujp** [reply]
I used a similar custom field I called `barcode` for years and was thinking to drop it and use the `mtn` field instead... It is when I realized there was no implementation for that at all. It does not make sens to do that for me, as I will have to maintain separate code anyway.
By experience, having as fewer plugins as possible saves me a lot of time in updates and maintenance. I thought I had an opportunity to do that but no...

**Reply by piloujp** [reply]
I just saw the PR, great! There was not much to update, and now everybody can use it.

**Reply by scottcwilson** [reply]
I wonder if we should add this to the list of fields checked during an admin search?

**Reply by piloujp** [reply]
It would be useful. One could use a barcode reader to directly fill the search form on categories/products page and get the product info to display quickly.

**Reply by piloujp** [reply]
I actually did this with older versions. In actual version, adding `'p.products_mpn',` around line 888 of `admin/category_product_listing.php` seems to be enough.

**Reply by scottcwilson** [reply]
OK. TY.

**Reply by piloujp** [reply]
Perhaps this same code should be added to `admin\includes\classes\observers\auto.search_box.php` too.

**Reply by drbyte** [reply]
#6812
